File: non-duplicate-array/NonDuplicateArray.swift
import Foundation

class NonDuplicate {
    func nonDuplicate(array: inout [Int]) -> [Int] {
        array.sort()
        var results: [Int] = []
        var i = 0
        while (i < array.count) {
            var j = i + 1
            var distinctIndex = i
            while(j < array.count) {
                if(array[distinctIndex] == array[j]) {
                    distinctIndex = j
                }
                j += 1
            }
            results.append(array[distinctIndex])
            i = distinctIndex
            i += 1
        }
        return results
    }
}

File: non-duplicate-array/NonDuplicateArray.java
import java.util.*;

public class NonDuplicateArray {
    public static Integer[] nonDuplicateArray(Integer[] array) {
        Arrays.sort(array);
        List<Integer> results = new ArrayList<>();
        for(int i = 0; i < array.length; i++) {
            int distinctIndex = i;
            boolean isDuplicated = false;
            for(int j = i + 1; j < array.length; j++) {
                if(isDuplicated && array[distinctIndex] != array[j]) { break; }
                if(array[i] == array[j]) {
                    distinctIndex = j;
                    isDuplicated = true;
                }
            }
            results.add(array[distinctIndex]);
            i = distinctIndex;
        }
        return results.toArray(new Integer[0]);
    }
}
